packages/xrpl/test/connection.test.ts

```diff
@@ -323,323 +507,484 @@ describe('Connection', function () {
+            } else {
+              resolve()
+            }
+          }
+        })
       })
 
-      breakConnection()
-    })
+      await breakConnection()
+      await connectedPromise
+    }, 70001)
   })
 
-  it('reconnect event on heartbeat failure', function (done) {
+  it('reconnect event on heartbeat failure', async () => {
     if (isBrowser) {
       if (navigator.userAgent.includes('PhantomJS')) {
         // inside PhantomJS this one just hangs, so skip as not very relevant
-        done()
         return
       }
     }
+
     // Set the heartbeat to less than the 1 second ping response
-    this.client.connection.config.timeout = 500
-    // Drop the test runner timeout, since this should be a quick test
-    this.timeout(5000)
-    // Hook up a listener for the reconnect event
-    this.client.connection.on('reconnect', () => done())
-    // Trigger a heartbeat
-    this.client.connection.heartbeat().catch((_error) => {
-      /* Ignore error */
+    // @ts-expect-error -- Testing private member
+    clientContext.client.connection.config.timeout = 500
+
+    const reconnectPromise = new Promise<void>((resolve) => {
+      // Hook up a listener for the reconnect event
+      clientContext.client.connection.on('reconnect', () => {
+        resolve()
+      })
     })
-  })
 
-  it('heartbeat failure and reconnect failure', function (done) {
+    // Trigger a heartbeat
+    try {
+      // @ts-expect-error -- Testing private member
+      await clientContext.client.connection.heartbeat()
+    } catch (_error) {
+      // ignore
+    }
+
+    await reconnectPromise
+  }, 5000)
+
+  it('heartbeat failure and reconnect failure', async () => {
     if (isBrowser) {
       if (navigator.userAgent.includes('PhantomJS')) {
         // inside PhantomJS this one just hangs, so skip as not very relevant
-        done()
         return
       }
     }
+
     // Set the heartbeat to less than the 1 second ping response
-    this.client.connection.config.timeout = 500
-    // Drop the test runner timeout, since this should be a quick test
-    this.timeout(5000)
+    // @ts-expect-error -- Testing private member
+    clientContext.client.connection.config.timeout = 500
     // fail on reconnect/connection
-    this.client.connection.reconnect = async (): Promise<void> => {
-      throw new XrplError('error on reconnect')
-    }
-    // Hook up a listener for the reconnect error event
-    this.client.on('error', (error, message) => {
-      if (error === 'reconnect' && message === 'error on reconnect') {
-        return done()
-      }
-      return done(new XrplError('Expected error on reconnect'))
+    jest
+      .spyOn(clientContext.client.connection, 'reconnect')
+      .mockImplementation(async (): Promise<void> => {
+        throw new XrplError('error on reconnect')
+      })
+
+    // clientContext?.client.connection.reconnect = async (): Promise<void> => {
+    //   throw new XrplError('error on reconnect')
+    // }
```

Comment:
Should this code be removed?

Reply:
Not sure

Reply:
It should either be uncommented or removed IMO